File: src/main/java/io/github/takenoha/towerdefense/persistence/TeamMutationResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;
import java.util.Optional;

/** Result of a team mutation, including the post-operation durable team when it still exists. */
public record TeamMutationResult(
        ManagementOutcome outcome,
        Optional<TeamRecord> team) {
    public TeamMutationResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(team, "team");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/TeamMutationResult.kt
package io.github.takenoha.towerdefense.persistence

import java.util.Optional
import kotlin.jvm.JvmRecord

/** Result of a team mutation, including the post-operation durable team when it still exists. */
@JvmRecord
data class TeamMutationResult(
    val outcome: ManagementOutcome,
    val team: Optional<TeamRecord>,
)
